lib/abstract_client.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const Beanstalk = require('fivebeans').client;
+const moment = require('moment');
+const bb = require('bluebird');
+const _ = require('lodash');
+
+/**
+ * Abstract beanstalk client class
+ */
+class AbstractClient {
+	constructor(config) {
+		this.client = null;
+		this.connected = false;
+		this.host = config.host?config.host : '127.0.0.1';
+		this.port = config.port?config.port : 11300;
+		this.tube = config.tube?config.tube : 'default';
+		this.enable_logging = config.enable_logging?config.enable_logging : false;
+	}
+
+	async start() {
+		const _this = this;
+		if (!_this.connected) {
+			return new Promise(resolve => {
+				_this.log(`connecting to beanstalkd at ${_this.host}:${_this.port}`);
+				
+				_this.client = new Beanstalk(_this.host, _this.port);
+				bb.promisifyAll(_this.client, {multiArgs: true});
+	
+				_this.client.on('connect', function () {
```

Comment:
and it's more robust to emit those event rather than silent them. it violates liskov substution of `SOLID` principle, `Postconditions cannot be weakened in a subtype.`